examples/Planning/ForecastReach.php

```diff
@@ -0,0 +1,389 @@
+ * locations and product codes, build a media plan, and generate a video ads reach forecast.
+ */
+class ForecastReach
+{
+    const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+
+    // US
+    const LOCATION_ID = '2840';
+
+    // 500,000
+    const BUDGET_MICROS = 500000000000;
+
+    const CURRENCY_CODE = 'USD';
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert
+        // them into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments([
+            ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT
+        ]);
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())
+            ->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    public static function runExample(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        self::showPlannableLocations($googleAdsClient);
+
+        self::showPlannableProducts($googleAdsClient, self::LOCATION_ID);
+
+        self::forecastManualMix(
+            $googleAdsClient,
+            $customerId,
+            self::LOCATION_ID,
+            self::CURRENCY_CODE,
+            self::BUDGET_MICROS
+        );
+
+        self::forecastSuggestedMix(
+            $googleAdsClient,
+            $customerId,
+            self::LOCATION_ID,
+            self::CURRENCY_CODE,
+            self::BUDGET_MICROS
+        );
+    }
```

Comment:
I think we're doing things a bit strange here.
LOCATION_ID and the other two are not open to be supplied by users.
So, why do we pass them here now? In other examples, for the real const like this, we just specify them where it's really used (e.g., PAGE_SIZE).

I don't quite get why we bother create another variables here.

Reply:
We should either open them to be specified by users (as optional arguments) (Preferred as it requires you to modify less, and it's more useful to users), or just remove all variables here.

Reply:
Agreed, I think the best would be to remove the useless parameters for these methods but keep the class constants for readability and flexibility (the constants are easy to find and edit when at the top of the file).

Reply:
Done for all methods but `requestReachCurve` because it is a utility method: I believe it makes better sense to keep it as-is.